src/Bridge/Doctrine/Orm/CollectionDataProvider.php

```diff
@@ -67,6 +68,9 @@ public function getCollection(string $resourceClass, string $operationName = nul
             }
         }
 
-        return $queryBuilder->getQuery()->getResult();
+        $query = $queryBuilder->getQuery();
+        //forces doctrine to not lazy load relations, see http://stackoverflow.com/questions/9848747/primary-key-of-owning-side-as-a-join-column
+        $query->setHint(Query::HINT_FORCE_PARTIAL_LOAD, true);
```

Comment:
I knew you had a solution, thanks I'm implementing this asap.
On Mon, 5 Sep 2016 at 13:26, Kévin Dunglas <notifications@github.com> wrote:

> In src/Bridge/Doctrine/Orm/CollectionDataProvider.php
> <https://github.com/api-platform/core/pull/717#discussion_r77508175>:
>
> > @@ -67,6 +68,9 @@ public function getCollection(string $resourceClass, string $operationName = nul
> >              }
> >          }
> >
> > -        return $queryBuilder->getQuery()->getResult();
> > +        $query = $queryBuilder->getQuery();
> > +        //forces doctrine to not lazy load relations, see http://stackoverflow.com/questions/9848747/primary-key-of-owning-side-as-a-join-column
> > +        $query->setHint(Query::HINT_FORCE_PARTIAL_LOAD, true);
>
> The other way is to add a public getResult(Query $query) method. It will
> allow users to decorate the class and change the behavior of this method.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/api-platform/core/pull/717/files/dd33381dd43ff75899b0f8a683c813b952f2ac41..179399d4fbbf3863f10d74911fdf8e921af56354#r77508175>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABQr849mFaTL9IePTPd0s82HvM5KovaAks5qm_xPgaJpZM4JxbFZ>
> .
>